File: app/src/main/java/com/ohmcoe/vultr/ServerList.java
package com.ohmcoe.vultr;


import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;


public class ServerList {

    private ArrayList<Server> serverLists;
    private String body;

    public ServerList() {

    }

    public List<Server> toList() {
        List<Server> list = serverLists;

        return list;
    }


    public ServerList(String response) {
        body = response;
        serverLists = new ArrayList<>();

        try {
            JSONObject jObj = new JSONObject(response);
            Iterator<?> keys = jObj.keys();

            while (keys.hasNext()) {
                String key = (String) keys.next();
                String result = jObj.getString(key);

                JSONObject serverJObj = new JSONObject(result);

                Server server = new Server();
                server.loadAttribute(serverJObj);

                serverLists.add(server);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

    public ArrayList<Server> getServerLists() {
        return serverLists;
    }


    public String getBody() {
        return body;
    }
}

File: app/src/main/java/com/ohmcoe/vultr/ServerList.kt
package com.ohmcoe.vultr


import org.json.JSONException
import org.json.JSONObject

import java.util.ArrayList


class ServerList {

    var serverLists: ArrayList<Server>? = null
    var body: String? = null

    constructor() {

    }

    fun toList(): List<Server> {
        val list = serverLists

        return list!!
    }


    constructor(response: String) {
        body = response
        serverLists = ArrayList<Server>()

        try {
            val jObj = JSONObject(response)
            val keys = jObj.keys()

            while (keys.hasNext()) {
                val result = jObj.getString(keys.next())

                val serverJObj = JSONObject(result)

                val server = Server()
                server.loadAttribute(serverJObj)

                serverLists!!.add(server)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }
}